### PR TITLE
Add test for exception handling on foreign threads.

### PR DESCRIPTION
**Pull request by adityamandaleeka**
Addresses #1794

### CODE REVIEW

File: tests/src/Exceptions/ForeignThread/ForeignThreadExceptionsNative.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include "stdio.h"
+
+#pragma warning(push)
+#pragma warning(disable:4265 4577)
+ #include <thread>
+#pragma warning(pop)
+
+
+#if defined _WIN32
+  #define DLL_EXPORT __declspec(dllexport)
```
Comment (tijoytom) [inline tests/src/Exceptions/ForeignThread/ForeignThreadExceptionsNative.cpp]: Can you include xplatform.h from Interop\Common instead of re-defining it

Comment (adityamandaleeka) [inline tests/src/Exceptions/ForeignThread/ForeignThreadExceptionsNative.cpp]: I've added it to Common/Platform/platformdefines.h and will use it from there instead.

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by wtgodbe** [comment]
LGTM

**Pull request merged**